demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lBytesWritten;
+
+    xResult = AzureIoTJSONWriter_Init( &xWriter, ucPropertyPayloadBuffer, sizeof( ucPropertyPayloadBuffer ) );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendBeginObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_BuilderBeginResponseStatus( &xAzureIoTHubClient,
+                                                                      &xWriter,
+                                                                      ( const uint8_t * ) sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT,
+                                                                      strlen( sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT ),
+                                                                      sampleazureiotPROPERTY_STATUS_SUCCESS,
+                                                                      ulVersion,
+                                                                      ( const uint8_t * ) sampleazureiotPROPERTY_SUCCESS,
+                                                                      strlen( sampleazureiotPROPERTY_SUCCESS ) );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendDouble( &xWriter, xUpdatedTemperature, sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_BuilderEndResponseStatus( &xAzureIoTHubClient,
+                                                                    &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendEndObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    lBytesWritten = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+    configASSERT( lBytesWritten > 0 );
+
+    LogDebug( ( "Sending acknowledged writable property. Payload: %.*s", lBytesWritten, ucPropertyPayloadBuffer ) );
+    xResult = AzureIoTHubClient_SendPropertiesReported( &xAzureIoTHubClient, ucPropertyPayloadBuffer, lBytesWritten, NULL );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "There was an error sending the reported properties: 0x%08x", xResult ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Property mesage callback handler
+ */
+static void prvHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                 void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    double xIncomingTemperature;
+    uint32_t ulVersion;
+    bool xWasMaxTemperatureChanged = false;
+
+    ( void ) pvContext;
+
+    LogDebug( ( "Property document payload : %.*s \r\n",
+                pxMessage->ulPayloadLength,
+                pxMessage->pvMessagePayload ) );
+
+    switch( pxMessage->xMessageType )
+    {
+        case eAzureIoTHubPropertiesGetMessage:
```

Comment:
Again not this PR or even this repo, but I think `eAzureIoTHubPropertiesGetMessage` really means `eAzureIoTHubPropertiesGetResponseMessage`?  As currently written with just "GetMessage" it looks like server is doing a get on the device itself rather than server respending to a device initiated get.